docs/examples/ex27.py

```diff
@@ -51,8 +52,8 @@ def acceleration_jacobian(u, du, v, dv, w):
        (w_j du_{i,j} + u_j dw_{i,j}) v_i
 
     """
-    return sum((np.einsum('j...,ij...->i...', w.w, du)
-                + np.einsum('j...,ij...->i...', u, w.dw)) * v)
+    return dot(np.einsum('j...,ij...->i...', w['w'], grad(u))
+               + np.einsum('j...,ij...->i...', u, grad(w['w'])), v)
```

Comment:
Do you think there should be some special shorthand in `skfem.helpers` for matrix-vector products? I think `dot` in Fenics works for that purpose but I don't like mixing "dot" products and "matrix-vector" or "matrix-matrix" products, somehow it doesn't make sense to me because I always like to think "dot product" as an inner product.

E.g. `skfem.helpers.mul` could be a new helper which detects the dimension of the inputs and then acts similarly as `dot` in Fenics (https://fenicsproject.org/qa/595/difference-inner-dot/).

Currently you can also write this as
```
ddot(prod(w, v), grad(u)) + ddot(prod(u, v), grad(w))
```
but it might not be too obvious what is the meaning.

Reply:
Actually here I like the einsum for its clarity. FEniCS gets this wrong in my opinion; it's a real muddle in UFL working out which way around the gradient of a vector field is.

I think it's O. K. to use dot between vectors and ddot between second order tensors as there's no chance of confusion there, but otherwise I'm never quite sure whether it's the first or last or inside indices that are being summed over. 

Also generally maybe it's clearer keeping the test function v separate?

Reply:
Sure I think it's also fine.